pulsar-client/src/main/java/org/apache/pulsar/client/impl/TableViewConfigurationData.java

```diff
@@ -29,13 +29,15 @@ public class TableViewConfigurationData implements Serializable, Cloneable {
 
     private String topicName = null;
     private long autoUpdatePartitionsSeconds = 60;
+    private String topicCompactionStrategy;
```

Comment:
How will the broker know which strategy should use? Or it just happened on the client side. For the broker side, we need to set the strategy by setting topic policies.

Reply:
I couldn't find a strong benefit to pass 'TopicCompactionStrategy' obj in this config.

I thought this config needs to be light-weight 'configs' instead of passing heavy objects.

Reply:
> How will the broker know which strategy should use? Or it just happened on the client side. For the broker side, we need to set the strategy by setting topic policies.

client can configure any matching topic compaction strategy if the user knows the compaction class name.

Right, on the broker side, ideally we can eventually set the compaction class name in the topic policy.

But for pip192 i think we could start with a simple hashmap (topicname-> compact strategy) for the new load balancer's internal system topic to use the compaction strategy.

Reply:
updated this name to `topicCompactionStrategyClassName`